eos/scattering/ee-to-ccbar.cc

```diff
@@ -0,0 +1,734 @@
+/*
+ * Copyright (c) 2021 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/scattering/ee-to-ccbar.hh>
+#include <eos/maths/complex.hh>
+#include <eos/utils/kinematic.hh>
+#include <eos/utils/kmatrix-impl.hh>
+#include <eos/utils/options.hh>
+#include <eos/utils/parameters.hh>
+#include <eos/utils/private_implementation_pattern-impl.hh>
+
+#include <vector>
+#include <array>
+#include <memory>
+
+namespace eos
+{
+
+    template <>
+    struct Implementation<EEToCCBar>
+    {
+        UsedParameter hbar;
+        UsedParameter alpha_em;
+        UsedParameter m_e;
+        UsedParameter m_D0;
+        UsedParameter m_D;
+        UsedParameter m_Dst0;
+        UsedParameter m_Dst;
+        UsedParameter m_Ds;
+        UsedParameter m_Dsst;
+
+        static const inline std::vector<std::string> resonance_names =
+        {
```

Comment:
To prevent future trouble:
- The corresponding enum should be placed direclty before the vector/string labels
- "Magic order" of elements should be commented on w/ disclaimer that they have to be kept in sync